### ISSUE TITLE
Can not reset values on elements when the "hardcoded" values changes 

### ISSUE THREAD

**Issue by ruthpozuelo** [microsoft/charticulator#722]
This is a bit hard to explain, but I will give it a go:
In the case below, I am using the measure "Legend exports vs. imports" to color the lines on the dumbell chart. If a new value appears or I change the measure to another one, the values will not reset to the new measure or in this case, they wont appear:
![image](https://user-images.githubusercontent.com/19476550/131645566-fde26091-5a7e-4d1b-82c7-9e8ae9df5caf.png)

As you can see, the data contains two values: Bigger exports and bigger imports but only bigger imports show after doing some changes (in this case removing blanks)
![image](https://user-images.githubusercontent.com/19476550/131645867-2ccab6e7-4d02-4e28-a9ab-14406a51bb01.png)

The only way I have found that fixes this, is to rebuild the chart from scratch. Here is the same chart once I rebuilt it:
![image](https://user-images.githubusercontent.com/19476550/131646507-7fd367f1-d0b0-4262-8df1-b5330659358c.png)

It happens also when I create a chart with filtered data, when I then remove the filters, the elements will show only for the original filtered data.
Hope I managed to explain!

**Comment by ruthpozuelo** [comment]
I discovered that by changing the value from "First" to "Last" all the objects appear again, in case it helps someone:
![image](https://user-images.githubusercontent.com/19476550/131652838-7d17a065-88da-4511-a907-a306e4251b0e.png)

**Comment by ruthpozuelo** [comment]
No, should I?

**Comment by zBritva** [comment]
No. I just ask details.  Aggregation function have sence when you group rows by some column. In another case it applies aggregation function for ONE row instead group of rows. So, when you don't have groupping First and Last function give the same result. And the same is for Avg, Max, Min, e.t.c. because they are appling for one one row and give the same value.

**Comment by zBritva** [comment]
The first part of question isn't clear to me, unfortunately. It seems you try to change column values and column names the same time and expect the visual handle it and update the Legend values.

**Comment by ruthpozuelo** [comment]
[Trade_public_SDMX - Copy (2).pbix.txt](https://github.com/microsoft/charticulator/files/7100358/Trade_public_SDMX.-.Copy.2.pbix.txt)

Sorry @zBritva , I should remember to attach the file so it is easier for you to troubleshoot. 
The viz in question (the rank chart) is still not working, so you can check how it has been built.

If I change from last to first different elements will be colored, which is odd...

I am noticing that when I color code elements with filtered data, and then in power bi I remove the filters, only the values that existed when I created the chart get displayed the other ones will get no color. Is it because the values are hardcoded?

let me know if I need to clarify anything else, will do my best!

**Comment by zBritva** [comment]
@ruthpozuelo Thanks! Scale values in "Legend" didn't update when new data is assigned.  In one of next version you will have option "Auto update values" to configure it:

![image](https://user-images.githubusercontent.com/10897951/131996386-9535125f-f828-4cbf-bbd9-1e1cf18595d9.png)

But you can't open the sccale editor from legend. To enable the property for your case you have to create invisible object.
Example:

1. Create "Symbol1" object and hide it:

![image](https://user-images.githubusercontent.com/10897951/131996716-9600a539-ff8f-4a01-a99e-3a3c15018ac8.png)

2. In scales panel you have scale mapping for "Legend1":

![image](https://user-images.githubusercontent.com/10897951/131996787-d284704a-b38a-4da7-a880-a26cdd25de32.png)

3. Drag & Drop this mapping to fill attribute of invisible "Symbol1":

![image](https://user-images.githubusercontent.com/10897951/131996870-78e84847-3d90-4a79-94b1-52cf56c66858.png)

4. Click on "Fill" attribute to open scale editor:

![image](https://user-images.githubusercontent.com/10897951/131997019-53dacf48-e497-41ab-b8a8-5ea02cb9133c.png)

5. You can remove the "Symbol1" object if you don't need it anymore or keep to modify scale again

@bongshin It seems we need scale editor for Legend objects too

**Comment by ruthpozuelo** [comment]
Fantastic @zBritva !!
I dont think I properly understand scales yet and that is why I keep getting into trouble...
I will give your proposal a go and thanks a million for the detailed instructions :)
